File: design-patterns/ts/abstract-factory/transport/two-wheel-transport-factory.ts
// two-wheel-transport-factory.ts

import AbstractTransportFactory from "./abstract-transport-factory";
import Bicycle from "./bicycle";
import Motorcycle from "./motorcycle";
import Transport from "./transport";

class TwoWheelTransportFactory implements AbstractTransportFactory {
    getTransport(type: string): (Transport | null) {
        if (type.toLowerCase() === "bicycle") {
            return new Bicycle();
        }

        if (type.toLowerCase() === "motorcycle") {
            return new Motorcycle();
        }

        return null;
    }
}

export default TwoWheelTransportFactory;

File: design-patterns/go/abstract_factory/transport/two_wheel_transport_factory.go
// two_wheel_transport_factory.go

package main

import (
	"errors"
	"strings"
)

type TwoWheelTransportFactory struct {
}

func NewTwoWheelTransportFactory() (twoWheelTransportFactory *TwoWheelTransportFactory) {
	twoWheelTransportFactory = &TwoWheelTransportFactory{}
	return
}

func (twoWheelTransportFactory *TwoWheelTransportFactory) GetTransport(identifier string) (Transport, error) {
	identifier = strings.ToLower(identifier)
	
	switch identifier {
		case "bicycle": return NewBicycle(), nil
		case "motorcycle": return NewMotorcycle(), nil
		default: return nil, errors.New("identifier does not match with any defined two wheel transport")
	}
}